### ISSUE TITLE
`onDownloadProgress` callback is triggered after calling `manager.pauseDownload`

### ISSUE THREAD

**Issue by rtritto** [theogravity/electron-dl-manager#8]
`main.ts`
```ts
ipcMain.on('download-pause', (_, id: string): void => {
  console.log('pauseDownload')
  manager.pauseDownload(id)
})

ipcMain.on('download-url', async (_, url: string) => {
  await manager.download({
    window,
    url,
    callbacks: {
      onDownloadProgress: ({ id, downloadRateBytesPerSecond, estimatedTimeRemainingSeconds, item, percentCompleted }) => {
        console.log('onDownloadProgress')
        ...
      }
    }
  })
})
```

Output:
```console
...
onDownloadProgress
onDownloadProgress
pauseDownload
onDownloadProgress  // it should no be called
```

**Comment by theogravity** [comment]
If you look at the code for how the pause is being called, it's directly on the DownloadItem itself. I don't think Electron / Chromium pauses immediately since it does continue to send progress after, and I think I've seen this behavior too elsewhere that I had to write a hack for it.

I can probably add something that signals back to the DownloadInitiator when pause is called....

**Comment by rtritto** [comment]
> I can probably add something that signals back to the DownloadInitiator when pause is called so when progress is emitted, it would just ignore it.

In the controller, the pause event should prevent `onDownloadProgress` to be triggered.

Note: idk the behavior of events

**Comment by rtritto** [comment]
Maybe using a `removeListener` on event of `onDownloadProgress`

**Comment by rtritto** [comment]
If the hack/fix creates problems/disadvantages, it can be rejected and close this issue.
So, in my case, I will check the current download state when the `onDownloadProgress` event is handled.

**Comment by theogravity** [comment]
I've published a draft: `electron-dl-manager@3.0.1-alpha.0`

Can you try that out and let me know if it works?


I've attached the code changes so you know what changed.

[Turn_off_progress_on_pause.patch.txt](https://github.com/user-attachments/files/16203493/Turn_off_progress_on_pause.patch.txt)

**Comment by rtritto** [comment]
It doesn't work (it's worse): after `pauseDownload`, `onDownloadProgress` increases by one the next time.

Some tests on same instance (each test use pause and then resume):

- First
```console
onDownloadProgress
onDownloadProgress
pauseDownload
onDownloadProgress
onDownloadProgress
```

```console
onDownloadProgress
onDownloadProgress
pauseDownload
onDownloadProgress
onDownloadProgress
onDownloadProgress
```

```console
onDownloadProgress
onDownloadProgress
pauseDownload
onDownloadProgress
onDownloadProgress
onDownloadProgress
onDownloadProgress
```
...

**Comment by theogravity** [comment]
I know what the issue is. I'm creating a new handler instead of referencing the existing one.

**Comment by theogravity** [comment]
Ok, try `electron-dl-manager@3.0.1-alpha.1`

**Comment by rtritto** [comment]
It works, thanks! 🎉

**Comment by theogravity** [comment]
Thanks. Published under `3.0.1`.
